legend-depot-artifacts-services/src/main/java/org/finos/legend/depot/services/artifacts/refresh/ProjectVersionRefreshHandler.java

```diff
@@ -304,6 +306,27 @@ private void updateProjectVersionData(StoreProjectData project, String versionId
         LOGGER.info("Finished updating project data [{}-{}-{}]", project.getGroupId(), project.getArtifactId(), versionId);
     }
 
+    private void updateProjectData(StoreProjectData projectData, String versionId)
+    {
+        if (!VersionValidator.isSnapshotVersion(versionId))
+        {
+            if (projectData.getLatestVersion() == null)
+            {
+                projectData.setLatestVersion(versionId);
```

Comment:
this code is a dup bellow. you can make more readable by splitting the calculation of what latest version shoudl be and then doing the actual set/update